build_tools/packaging/linux/build_package.py

```diff
@@ -0,0 +1,497 @@
+    if config.enable_rpath:
+        print("ENABLE RPATH")
+        subprocess.run(["python3", "runpath_to_rpath.py", package_dir])
+
+    package_with_dpkg_deb(package_dir, config.pkg_dir, debpkg_name)
+
+def generate_contol_file(pkginfo, package_dir, config: PackageConfig):
+    '''Function will generate control file for debian package'''
+    print("Generate control file")
+    control_dir = f"{package_dir}/DEBIAN"
+    os.makedirs(control_dir, exist_ok=True)
+    controlfile = f"{control_dir}/control"
+
+    pkg_name = update_package_name(pkginfo.get("Package"), config )
+    # Only required for debian developement package
+    pkg_name = pkg_name.replace("-devel", "-dev")
+    arch = pkginfo.get("Architecture")
+    description = pkginfo.get("Description")
+    version = pkginfo.get("Version")
+    section = pkginfo.get("Section")
+    priority = pkginfo.get("Priority")
+    maintainer = pkginfo.get("Maintainer")
+    homepage = pkginfo.get("Homepage")
+    depends_list = pkginfo.get("DEBDepends", [])
+    depends = convert_to_versiondependency(depends_list, config)
+    # Note: The dev package name update should be done after version dependency
+    # Package.json maintains development package name as devel
+    depends = depends.replace("-devel", "-dev")
+
+    with open(controlfile, 'w') as f:
+        f.write(f"Architecture: {arch}\n")
+        f.write(f"Depends: {depends}\n")
+        f.write(f"Description: {description}\n")
+        f.write(f"Homepage: {homepage}\n")
+        f.write(f"Maintainer: {maintainer}\n")
+        f.write(f"Package: {pkg_name}\n")
+        f.write(f"Priority: {priority}\n")
+        f.write(f"Section: {section}\n")
+        f.write(f"Version: {version}\n")
+        f.close()
+
+def copy_package_contents(source_dir, destination_dir):
+    ''' Copy package contents from artfactory to package directory'''
+   
+    if not os.path.isdir(source_dir):
+        print(f"Directory does not exist: {source_dir}")
+        return
+
+    # Ensure destination directory exists
+    os.makedirs(destination_dir, exist_ok=True)
+
+    # Copy each item from source to destination
+    for item in os.listdir(source_dir):
+        s = os.path.join(source_dir, item)
+        d = os.path.join(destination_dir, item)
+        if os.path.isdir(s):
+            shutil.copytree(s, d, dirs_exist_ok=True)
+        else:
+            shutil.copy2(s, d)
+
+def package_with_dpkg_deb(source_dir, output_dir, package_name):
+    ''' Create deb package '''
+    # Construct paths
+    output_deb = f"{output_dir}/{package_name}.deb"
+
+    # Build the command
+    cmd = [
+        "fakeroot",
+        "dpkg-deb",
+        "-Zgzip",
+        "--build",
+        source_dir,
+        output_deb
+    ]
+
+    # Execute the command
+    try:
+        subprocess.run(cmd, check=True)
+        print("Package built successfully.")
+    except subprocess.CalledProcessError as e:
+        print("Error building package:", e)
```

Comment:
probably should pass an error code up to the caller (and eventually to the executor)

Reply:
Ack